Sources/PostgresNIO/New/PSQLRowSequence.swift

```diff
@@ -0,0 +1,607 @@
+import NIOCore
+import NIOConcurrencyHelpers
+
+#if swift(>=5.5) && canImport(_Concurrency)
+struct PSQLRowSequence: AsyncSequence {
```

Comment:
Can you add a comment here explaining the use of the "value type wrapping a consumer with a reference type" pattern? You don't have to go and write one of those 5-page rambles the way I sometimes do (in fact, please, for the love of everything you hold dear- _don't_ do that! 😆), a couple lines would probably more than so.

Reply:
Short story is to prepare for eventual move only types. We only use the reference type internally to signal when the Sequence was released. Should I add this in code?

Reply:
I wouldn't mind a short comment about it, don't need to get into detail.